### PR TITLE
libint 2.7.2 (new formula)

### PR DESCRIPTION
**Pull request by junghans**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
For #132494

### CODE REVIEW

File: Formula/libint.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "boost"
+  depends_on "mpfr"
+  depends_on "python@.3.11"
+
+  def install
+    system "./configure", *std_configure_args
+    system "make"
+    system "make", "install"
+  end
```
Comment (chenrui333) [inline Formula/libint.rb:16]: we need test for this

File: Formula/libint.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Libint < Formula
+  desc "A library for computing electron repulsion integrals efficiently"
```
Comment (chenrui333) [inline Formula/libint.rb:2]:
```suggestion
  desc "Library for computing electron repulsion integrals efficiently"
```

File: Formula/libint.rb (hunk 1 above)
Comment (junghans) [reply Formula/libint.rb:16]: Yeah I was looking at https://github.com/evaleev/libint/blob/master/tests/hartree-fock/hartree-fock.cc, but it is so long!

@evaleev, do you a short example to test if libint works? 

Something like:
https://github.com/Homebrew/homebrew-core/blob/5d9c4b3f2496856471ab9c1cda318c4ad80f956a/Formula/fftw.rb#L74-L94

File: Formula/libint.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Libint < Formula
+  desc "Library for computing electron repulsion integrals efficiently"
+  homepage "https://github.com/evaleev/libint"
+  url "https://github.com/evaleev/libint/archive/v2.7.2.tar.gz"
+  sha256 "fd0466ce9eb6786b8c5bbe3d510e387ed44b198a163264dfd7e60b337e295fd9"
+  license "GPL-2.0-or-later"
```
Comment (chenrui333) [inline Formula/libint.rb:6]: this needs to be `license all_of: ["GPL-2.0-or-later", "GPL-3.0-or-later"]`

File: Formula/libint.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "boost"
```
Comment (chenrui333) [inline Formula/libint.rb:10]: it should also depends on "eigen"?
```suggestion
  depends_on "boost"
  depends_on "eigen"
```


```
/opt/homebrew/Cellar/libint/2.7.2/include/libint2/./engine.impl.h:31:10: fatal error: 'Eigen/Core' file not found
#include <Eigen/Core>
         ^~~~~~~~~~~~
1 error generated.
```

File: Formula/libint.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/libint.rb]: I was wrong with this comment, this actually should be LGPL3+ and GPL3+, just fixed it.

**Comment by chenrui333** [comment]
Just updated the license and amended the commit the message. Good to go.

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**